artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -576,15 +612,17 @@ public QueueImpl(final long id,
       if (this.addressInfo != null && this.addressInfo.isPaused()) {
          this.pause(false);
       }
+
+      this.ringSize = ringSize == null ? (addressSettingsRepository != null ? addressSettingsRepository.getMatch(address.toString()).getDefaultRingSize() : -1) : ringSize;
    }
 
    // Bindable implementation -------------------------------------------------------------------------------------
 
    @Override
    public boolean allowsReferenceCallback() {
-      // non descructive queues will reuse the same reference between multiple consumers
+      // non destructive queues will reuse the same reference between multiple consumers
       // so you cannot really use the callback from the MessageReference
-      return !nonDestructive;
+      return !isNonDestructive();
```

Comment:
? Why the chang. Its internal field and allows access by field reference

Reply:
(nit)

Reply:
I remember changing this, but I don't remember why. :)  I'll change them back.